### PR TITLE
[JENKINS-44014] - Fix the reported runtime exceptions (Pipeline integration)

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This PR fixes two issues reported in [JENKINS-44014](https://issues.jenkins-ci.org/projects/JENKINS/issues/JENKINS-44014). I assume both issues are related to the newly introduced Pipeline support, hence @reviewbybees 

* Prevent class cast exception and add diagnostics if the `ownerTask` of Pipeline placeholder is not a Job. Multibranch?
* Prevent NullPointerException in `AdvancedQueueSorter#onLeft()` when item is missing in the queue

CC @jglick who commented to this pull request

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
As a current plugin maintainer, I officially give up on maintaining spaces/tabs in this code

File: src/main/java/jenkins/advancedqueue/ItemTransitionLogger.java [hunk 1]
```diff
@@ -37,27 +38,27 @@
...
 		LOGGER.fine("Buildable: " + info.toString());
 	}
 
-	static public void logStartedItem(ItemInfo info) {
+	static public void logStartedItem(@Nonnull ItemInfo info) {
 		LOGGER.fine("Starting: " + info.toString());
 	}
 
-	static public void logCanceledItem(ItemInfo info) {
+	static public void logCanceledItem(@Nonnull ItemInfo info) {
```
Comment (svanoort) [inline src/main/java/jenkins/advancedqueue/ItemTransitionLogger.java]: Like it!

File: src/main/java/jenkins/advancedqueue/sorter/SorterStrategy.java [hunk 2]
```diff
@@ -54,21 +55,21 @@ public SorterStrategyDescriptor getDescriptor() {
 	 *            the weight before returning
 	 * @return the {@link SorterStrategyCallback} provided to the call must be returned
 	 */
-	public abstract SorterStrategyCallback onNewItem(Queue.Item item, SorterStrategyCallback weightCallback);
+	public abstract SorterStrategyCallback onNewItem(@Nonnull Queue.Item item, SorterStrategyCallback weightCallback);
```
Comment (svanoort) [inline src/main/java/jenkins/advancedqueue/sorter/SorterStrategy.java]: Can the callback be null, just out of curiosity?

**Comment by oleg-nenashev** [comment]
> Can the callback be null, just out of curiosity?

Didn't touch it, likely not. This plugin codebase is not annotated, and it is hard to do anything without a "total annotation" pull request like I did for several plugins before.

@reviewbybees done

**Comment by oleg-nenashev** [comment]
I am going to merge and release it. At least it will unlikely make the things worse than they were before

**Comment by oleg-nenashev** [comment]
@jglick 
So it comes from here (below is a new code with the check):

```
PriorityConfigurationCallback getPriority(@Nonnull ExecutorStepExecution.PlaceholderTask task, @Nonnull PriorityConfigurationCallback priorityCallback) {
        Queue.Task ownerTask = task.getOwnerTask();
          if (ownerTask instanceof Job<?, ?>) {
...
```

When we try to retrieve placeholder of the priority task, it fails in the previous code which assumed that `ownerTask` is always a Job. Not sure why it's not in this case, MultiBranch? I have not debugged it, but we will have better logging in new versions at least.

**Comment by jglick** [comment]
So this is intended as a hotfix, not the final fix, since we still do not know what the problematic `Task` implementation actually was.

**Comment by jglick** [comment]
Actually I think I [see the problem](https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/14ecae6c243b892a44317c337f467e233053ed0b/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java#L316-L319): a null `run()`, presumably due to a build with corrupt metadata of some kind. In which case quietly ignoring a non-`Job` is the correct behavior.

**Pull request merged**